src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -117,25 +120,32 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseSer
      */
     @GetMapping("/complaints/result/{resultId}")
     @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    // TODO: the URL should rather be "/complaints?resultId={resultId}" and resultId should be mandatory
     public ResponseEntity<Complaint> getComplaintByResultId(@PathVariable Long resultId) {
         log.debug("REST request to get Complaint associated to result : {}", resultId);
-        Optional<Complaint> complaint = complaintService.getByResultId(resultId);
-
-        if (complaint.isPresent() && complaint.get().getResult() != null) {
-            var participation = (StudentParticipation) complaint.get().getResult().getParticipation();
-            Exercise exercise = participation.getExercise();
-            if (!authCheckService.isAtLeastInstructorForExercise(exercise)) {
-                complaint.get().getResult().setAssessor(null);
-
-                if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
-                    // filter student information if user is not instructor but at least teaching assistant (means that user is teaching assistant)
-                    complaint.get().filterSensitiveInformation();
-                    participation.filterSensitiveInformation();
-                }
-            }
+        var optionalComplaint = complaintService.getByResultId(resultId);
+        if (optionalComplaint.isEmpty()) {
+            return notFound();
```

Comment:
```suggestion
            return ResponseEntity.ok().build();
```

If there is no complaint for a result, the frontend is expecting a response with status code 200 and an empty body, not a 404 response. This behavior is also specified in the JSDoc string. The frontend is using this endpoint both to retrieve a complaint for a certain result and to check whether such a complaint exists in the first place.

Since the non-existence of a complaint for a result is not an unexpected error in this case, this should of course not be relayed to the user, which happens at the moment as @HanyaElhashemy noticed while testing. See screenshot below:

![OnPaste 20191216-171636](https://user-images.githubusercontent.com/7109225/70923220-d3f3b900-2027-11ea-9f53-4dfbb17030bd.png)

I'm not sure which approach conforms better with REST standards when the API user knows that the resource's existence is optional. I guess a 404 response would be more explicit than an empty body but than the frontend needs to handle this "error". In this particular case, a 404 response could also be used to convey that the result for which the complaint is requested does not exist, so its meaning is a bit ambivalent here anyway.

Long story short, I think the best course of action right now is probably to simply restore the previous behavior and return an empty body if there is no complaint (see code suggestion).